driver/src/orderbook/streamed/state.rs

```diff
@@ -186,6 +212,61 @@ impl State {
         balance.balance += event.burnt_fees;
         Ok(())
     }
+
+    fn apply_trade(
+        &mut self,
+        user: UserId,
+        order: OrderId,
+        executed_sell_amount: u128,
+        executed_buy_amount: u128,
+        batch_id: BatchId,
+        revert: bool,
+    ) -> Result<(), Error> {
+        let order = self
+            .orders
+            .get_mut(&(user, order))
+            .ok_or(Error::UnknownOrder)?;
+
+        let sell_token = self
+            .tokens
+            .address(order.sell_token)
+            .ok_or(Error::UnknownToken)?;
+        let buy_token = self
+            .tokens
+            .address(order.buy_token)
+            .ok_or(Error::UnknownToken)?;
+
+        if order.has_limited_amount() {
+            if revert {
+                order.used_amount -= executed_sell_amount;
+            } else {
+                order.used_amount += executed_sell_amount;
+            }
+        }
+        self.apply_balance_from_trade(user, sell_token, executed_sell_amount, batch_id, !revert);
+        self.apply_balance_from_trade(user, buy_token, executed_buy_amount, batch_id, revert);
+        Ok(())
+    }
+
+    fn apply_balance_from_trade(
+        &mut self,
+        user: UserId,
+        token: TokenAddress,
+        executed_amount: u128,
+        batch_id: BatchId,
+        subtract: bool,
+    ) {
+        let balance = self.balances.entry((user, token)).or_default();
+        balance.update_deposit_balance(batch_id);
```

Comment:
Here again I think it would be nice if balance itself would decide when to update deposits.